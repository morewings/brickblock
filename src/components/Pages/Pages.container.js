import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {setCurrentPage} from 'Redux/actions';
import Pages from 'components/Pages/Pages';

function mapStateToProps(state, props) {
  return {
    dataLength: state.contributors[props.dataSource].raw.length,
    currentPage: state.pagination[props.dataSource].page,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({setCurrentPage}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Pages);
