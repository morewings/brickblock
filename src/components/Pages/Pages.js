import React from 'react';
import PropTypes from 'prop-types';
import {Pagination, PaginationItem, PaginationLink} from 'reactstrap';
import config from 'config';

const Pages = (props) => {
  const pagesAmount = Math.ceil(props.dataLength / config.itemsPerPage);
  const pagesTemplate = new Array(pagesAmount).fill('');
  const onClick = page => (e) => {
    e.preventDefault();
    props.setCurrentPage(page, props.dataSource);
  };
  return (
    <Pagination>
      <PaginationItem disabled={props.currentPage === 1}>
        <PaginationLink
          onClick={onClick(props.currentPage - 1)}
          previous
          href="#" />
      </PaginationItem>
      {pagesTemplate.map((empty, i) => (
        <PaginationItem
          active={props.currentPage === (i + 1)}
          key={`page-${i}`}>
          <PaginationLink
            onClick={onClick(i + 1)}
            href="#">
            {i + 1}
          </PaginationLink>
        </PaginationItem>
        ))}
      <PaginationItem disabled={props.currentPage === pagesAmount}>
        <PaginationLink
          onClick={onClick(props.currentPage + 1)}
          next
          href="#" />
      </PaginationItem>
    </Pagination>
  );
};

Pages.propTypes = {
  dataLength: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  setCurrentPage: PropTypes.func.isRequired, // eslint-disable-line
  dataSource: PropTypes.string.isRequired, // eslint-disable-line
};

export default Pages;
