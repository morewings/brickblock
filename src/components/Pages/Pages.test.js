import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Pages from 'components/Pages/Pages';


describe('components > Pages > Pages', () => {
  beforeEach(() => {
    // sample.mockClear();
  });
  it('renders without crashing', () => {
    const wrapper = shallow(<Pages />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
