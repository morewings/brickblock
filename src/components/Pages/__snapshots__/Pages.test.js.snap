// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Pages > Pages renders without crashing 1`] = `
<Pagination
  tag="ul"
>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      previous={true}
      tag="a"
    />
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      tag="a"
    >
      1
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      tag="a"
    >
      2
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      tag="a"
    >
      3
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      tag="a"
    >
      4
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      tag="a"
    >
      5
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    tag="li"
  >
    <PaginationLink
      href="#"
      next={true}
      tag="a"
    />
  </PaginationItem>
</Pagination>
`;
